File: client-java/controller/src/main/java/org/evomaster/client/java/controller/mongo/selectors/LessThanEqualsSelector.java
package org.evomaster.client.java.controller.mongo.selectors;

import org.evomaster.client.java.controller.mongo.operations.LessThanEqualsOperation;
import org.evomaster.client.java.controller.mongo.operations.QueryOperation;

import java.util.Objects;

/**
 * { field: { $lte: value } }
 */
public class LessThanEqualsSelector extends SingleConditionQuerySelector {

    public static final String LESS_THAN_EQUALS = "$lte";

    @Override
    protected QueryOperation parseValue(String fieldName, Object value) {
        Objects.requireNonNull(fieldName);
        if (value==null) {
            return null;
        } else {
            return new LessThanEqualsOperation<>(fieldName, value);
        }
    }

    @Override
    protected String operator() {
        return LESS_THAN_EQUALS;
    }
}

File: client-java/controller/src/main/java/org/evomaster/client/java/controller/mongo/selectors/LessThanEqualsSelector.kt
package org.evomaster.client.java.controller.mongo.selectors

import org.evomaster.client.java.controller.mongo.operations.LessThanEqualsOperation
import org.evomaster.client.java.controller.mongo.operations.QueryOperation

/**
 * { field: { $lte: value } }
 */
class LessThanEqualsSelector : SingleConditionQuerySelector() {
    override fun operator(): String {
        return "\$lte"
    }

    override fun parseValue(fieldName: String, value: Any?): QueryOperation = LessThanEqualsOperation(fieldName,value)
}